src/main/java/uk/gov/justice/hmpps/prison/service/OffenderTransactionHistoryService.java

```diff
@@ -0,0 +1,61 @@
+package uk.gov.justice.hmpps.prison.service;
+
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.validation.annotation.Validated;
+import uk.gov.justice.hmpps.prison.api.model.OffenderTransactionHistoryDto;
+import uk.gov.justice.hmpps.prison.repository.OffenderTransactionHistoryRepository;
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderTransactionHistory;
+import uk.gov.justice.hmpps.prison.service.transformers.OffenderTransactionHistoryTransformer;
+import uk.gov.justice.hmpps.prison.values.AccountCode;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+@Service
+@Transactional(readOnly = true)
+@AllArgsConstructor
+@Validated
+@Slf4j
+public class OffenderTransactionHistoryService {
+
+    private OffenderTransactionHistoryRepository repository;
+
+    public List<OffenderTransactionHistoryDto> getTransactionHistory(final Long offenderId,
+                                                                     final Optional<String> accountCode,
+                                                                     final LocalDate fromDate,
+                                                                     final LocalDate toDate) {
+
+        checkNotNull(offenderId, "offender-id can't be null");
```

Comment:
Not sure this can ever be null unless used outside of Controller/Resource.

Reply:
indeed this is there to protect the service code and to communicate to future developers that they can't pass null for this

Reply:
I see your point but seems overkill if you're using an Optional?

Reply:
thanks Mike, not using optional for offenderId though